Tests/Indicators/ImpliedVolatilityTests.cs

```diff
@@ -44,77 +46,153 @@ public void SetUp()
             DividendYieldUpdatesPerIteration = 1;
         }
 
-        // For comparing IB's value
-        [TestCase("SPX230811C04300000", 0.2)]
-        [TestCase("SPX230811C04500000")]
-        [TestCase("SPX230811C04700000")]
-        [TestCase("SPX230811P04300000", 0.015)]
-        [TestCase("SPX230811P04500000")]
-        [TestCase("SPX230811P04700000", 0.075)]
-        [TestCase("SPX230901C04300000")]
-        [TestCase("SPX230901C04500000")]
-        [TestCase("SPX230901C04700000")]
-        [TestCase("SPX230901P04300000")]
-        [TestCase("SPX230901P04500000")]
-        [TestCase("SPX230901P04700000")]
-        [TestCase("SPY230811C00430000", 0.055)]
-        [TestCase("SPY230811C00450000", 0.015)]
-        [TestCase("SPY230811C00470000")]
-        [TestCase("SPY230811P00430000", 0.015)]
-        [TestCase("SPY230811P00450000")]
-        [TestCase("SPY230811P00470000", 0.075)]
-        [TestCase("SPY230901C00430000", 0.025)]
-        [TestCase("SPY230901C00450000")]
-        [TestCase("SPY230901C00470000")]
-        [TestCase("SPY230901P00430000")]
-        [TestCase("SPY230901P00450000")]
-        [TestCase("SPY230901P00470000", 0.035)]
-        public void ComparesAgainstExternalData(string fileName, double errorMargin = 0.01, int column = 2)
+        [TestCase(7, 6)]
+        public void ComparesAgainstExternalData1(int callColumn, int putColumn, double errorMargin = 0.08)
         {
-            var path = Path.Combine("TestData", "greeks", $"{fileName}.csv");
-            var symbol = ParseOptionSymbol(fileName);
-            var underlying = symbol.Underlying;
+            var path = Path.Combine("TestData", "greeksindicator");
+            foreach (var style in new string[] { "european", "american" })
+            {
+                var file = Path.Combine(path, style, "third_party_1_greeks.csv");
+
+                foreach (var line in File.ReadAllLines(file).Skip(3))
+                {
+                    var items = line.Split(',');
+
+                    var interestRate = decimal.Parse(items[^2]);
+                    var dividendYield = decimal.Parse(items[^1]);
+
+                    var model = ParseSymbols(items, file.Contains("american"), out var call, out var put);
+
+                    var callIndicator = new ImpliedVolatility(call, put, interestRate, dividendYield, model);
+                    var putIndicator = new ImpliedVolatility(put, call, interestRate, dividendYield, model);
+
+                    RunTestIndicator(call, put, callIndicator, putIndicator, items, callColumn, putColumn, errorMargin);
+                }
+            }
+        }
+
+        // We are unsure about the smoothing function
+        [TestCase(7, 6)]
+        public void ComparesAgainstExternalData2(int callColumn, int putColumn, double errorMargin = 10)
```

Comment:
Isn't an error margin of 10 too permissive?

Reply:
Hey! We just place it there as a reference, but we are not going to spend time to reverse engineer the smoothing method